src/common/state/DBOptionsAttributes.C

```diff
@@ -1300,7 +1300,13 @@ DBOptionsAttributes::SetInt(const std::string &name, int defaultValue)
         optInts.push_back(defaultValue);
     }
     else
-        optInts[bIndex] = defaultValue;
+    {
```

Comment:
> it seems like we're adjusting a perfectly fine and working class

I rescind my opinion that `DBOptionsAttributes()` is *prefectly fine  and working class*...I don't think it behaves well in the presence of *mistakes* made by the caller. I mean, it may not segv if you call `SetInt("foa", 5)` for an instances that has an `Int` option named `foo` you are trying to set but it sure won't do the right thing either :wink:

Reply:
Yes, this is all to accommodate libsim's VisIt_OptionList_setValueI/S calls for Enum database read options, which currently crashes.

Calling setValueI('someoption', 2) where 'someoption' is declared an Enum in the DBOptionsAttributes ends up calling DBOptionsAttributes::SetInt('someoption', 2). This method calls FindIndex to find the index of 'someoption', which returns a valid index into the Enum options list, but then attempts to use that index to set a value in the Int options list, which doesn't exist and segv's.



Can't really create VisIt_OptionList_setValueE (for Enum) either implemented as String or Int because it equates to the same native data type as setValueI or setValueS methods and causes compile error regarding overrides that cannot be resolved. Sorry I cannot point right now to the exact source file where the compile error occurs, I somehow lost my stash of changes where I tried the setValueE route.

Without the change, sim codes cannot attempt to set  Read options for readers that use Enum read options, because it causes a crash.

Reply:
So I just remembered, the original bug ticket is for Write options (eg used for an Export of the database).